### PR TITLE
fix(circuit): disable circuit breaker in development

### PR DESCRIPTION
**Pull request by nellyk**
## Description

Disabling the circuit breaker in development environment
## Motivation and Context

Currently during local development the circuit breaker opens up and causes a lot of log generation. This is usually useful but for the development environment it doesn't provide for a seamless experience when creating experiences 

## How Has This Been Tested?

Unit tests have been added
## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

NA

### CODE REVIEW

**Review by JAdshead — DISMISSED** [review]

File: docs/api/modules/App-Configuration.md [hunk 1]
```diff
@@ -452,6 +452,8 @@ if (!global.BROWSER) {
 
 The `eventLoopDelayThreshold` directive accepts a number representing the threshold of the event loop delay (in milliseconds) before opening the circuit. Once the circuit is open, it will remain open for 10 seconds and close at that time pending the event loop delay. The default value is `250`. If you desire to disable the event loop delay potion of the circuit breaker, set this value to `Infinity`. The circuit will also open if the error rate exceeds 10%. In practice, `eventLoopDelayThreshold` allows for tuning server side rendering (SSR) of Modules. We may increase request throughput by temporarily disabling SSR at high load through event loop delay monitoring.
 
+> This is disabled in when NODE_ENV=development
```
Comment (mtomcal) [inline docs/api/modules/App-Configuration.md:455]:
```suggestion
> This is disabled when NODE_ENV=development
```

File: __tests__/server/utils/createCircuitBreaker.spec.js [hunk 2]
```diff
@@ -64,6 +65,19 @@ describe('Circuit breaker', () => {
     expect(value).toBe(true);
   });
 
+  it('should not open the circuit when in development environment', async () => {
+    process.env.NODE_ENV = 'development';
+    expect.assertions(2);
+    setEventLoopDelayThreshold(-1);
```
Comment (mtomcal) [inline __tests__/server/utils/createCircuitBreaker.spec.js:71]: This is to flip the switch on the circuit breaker every time right?

**Review by mtomcal — CHANGES_REQUESTED** [review]

**Review by Francois-Esquire — DISMISSED** [review]

File: __tests__/server/utils/createCircuitBreaker.spec.js (hunk 2 above)
Comment (nellyk) [reply __tests__/server/utils/createCircuitBreaker.spec.js:71]: yes it is, so that we can ensure it is would be running

**Comment by nellyk** [comment]
autocannon tests 

Before this changes

Screenshot 2020-09-18 at 14 15 06


With this changes
Screenshot 2020-09-18 at 14 11 13

File: src/server/utils/createCircuitBreaker.js [hunk 3]
```diff
@@ -43,6 +43,7 @@ eventLoopDelayHistogram.enable();
 
 const checkMaxEventLoopDelay = async () => {
   // Return if root module is not loaded, as that is where threshold is configured
+  if (process.env.NODE_ENV === 'development') return;
   if (!getModule(rootModuleName)) return;
```
Comment (JAdshead) [inline src/server/utils/createCircuitBreaker.js:47]:
```suggestion
  if (process.env.NODE_ENV === 'development') return;
  // Return if root module is not loaded, as that is where threshold is configured
  if (!getModule(rootModuleName)) return;
```

**Review by mtomcal — DISMISSED** [review]

**Review by JAdshead — CHANGES_REQUESTED** [review]

**Review by Francois-Esquire — DISMISSED** [review]

**Review by 10xLaCroixDrinker — DISMISSED** [review]

**Review by infoxicator — DISMISSED** [review]

**Review by JamesSingleton — DISMISSED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by JamesSingleton — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

**Pull request merged**